### ISSUE TITLE
Pod Update Should Update Default Subspec

### ISSUE THREAD

**Issue by Kurry** [CocoaPods/CocoaPods#4179]
I ran into this issue: https://github.com/CocoaPods/CocoaPods/issues/2209 and think it's a very confusing behavior. I think if you run `pod update Pod`, it should default to updating the default subspec, which is similar to the fact that if you don't specify a subspec it installs the default. Right?

**Comment by segiddins** [comment]
Can you please provide more information, such as your version of CocoaPods, your Podfile, the command you ran, what you expected to happen, and what happened instead?

**Comment by Kurry** [comment]
### Cocoapods Version 0.37.2
### Podfile
```
new-host:Example ktran$ cat Podfile
target 'PINRemoteImage', :exclusive => true do
  pod "PINRemoteImage/Core", :path => "../"
  pod "PINCache"
  pod "libwebp"
end

target 'PINRemoteImage Tests', :exclusive => true do
  pod "PINRemoteImage/Core", :path => "../"
end
```

### Podspec 
```
Pod::Spec.new do |s|
  s.name             = "PINRemoteImage"
  s.version          = "1.1.2"
  s.summary          = "A thread safe, performant, feature rich image fetcher"
  s.homepage         = "https://github.com/pinterest/PINRemoteImage"
  s.license          = 'Apache 2.0'
  s.author           = { "Garrett Moon" => "garrett@pinterest.com" }
  s.source           = { :git => "https://github.com/pinterest/PINRemoteImage.git", :tag => "1.1.2" }
  s.social_media_url = 'https://twitter.com/garrettmoon'

  s.platform     = :ios, '6.0'
  s.requires_arc = true
  
  # Exclude optional FLAnimatedImage module
  s.default_subspec = 'Core'
  
  ### Subspecs
  s.subspec 'Core' do |cs|
    cs.source_files = 'Pod/Classes/**/*.{h,m}'
    cs.exclude_files = 'Pod/Classes/Image Categories/FLAnimatedImageView+PINRemoteImage.h', 'Pod/Classes/Image Categories/FLAnimatedImageView+PINRemoteImage.m'
    cs.public_header_files = 'Pod/Classes/**/*.h'
    cs.frameworks = 'UIKit', 'ImageIO', 'CoreImage'
    cs.dependency 'PINCache', '>=2.0'
  end

  s.subspec "FLAnimatedImage" do |fs|
    fs.dependency 'PINRemoteImage/Core'
    fs.source_files = 'Pod/Classes/Image Categories/FLAnimatedImageView+PINRemoteImage.h', 'Pod/Classes/Image Categories/FLAnimatedImageView+PINRemoteImage.m'
    fs.dependency 'FLAnimatedImage', '>= 1.0'
  end
end
```

### Command
```
new-host:Example ktran$ bundle exec pod update PINRemoteImage
[!] The `PINRemoteImage` Pod is not installed and cannot be updated
```
I expected the update command to update the `PINRemoteImage/Core` dependency in my `Podfile`. Instead, I received the error above.

**Comment by Kurry** [comment]
@segiddins After you look at this, if you could just point me to the files I would need to look at, I'd gladly make a pull request to fix this.

**Comment by segiddins** [comment]
Super weird, I could've sworn I'd fixed this at some point.

**Comment by segiddins** [comment]
Was done via https://github.com/CocoaPods/CocoaPods/commit/3cd3de4052deb6b13f5bab6545ba62bab005f097, which shipped in 0.38.
